File: 백준/Silver/1260. DFS와 BFS/DFS와 BFS.cc
#include<iostream>
#include<queue>
#include<vector>
#include<algorithm>
#include<cstring>
using namespace std;

int N, M, s;
vector<int> graph[1005];
bool vis[1005];

void DFS(int cur) {
	vis[cur] = true;
	cout << cur << ' ';
	for (auto a : graph[cur]) {
		if (vis[a])continue;
		DFS(a);
	}
}

void BFS() {
	queue<int> q;
	q.push(s);
	vis[s] = true;
	while (!q.empty()) {
		int cur = q.front();
		cout << cur << ' ';
		q.pop();

		for (auto a : graph[cur]) {
			if (vis[a])continue;
			q.push(a);
			vis[a] = true;
		}
	}
}

int main(void) {
	cin.tie(NULL)->sync_with_stdio(false);

	cin >> N >> M >> s;
	for (int m = 0; m < M; ++m) {
		int u, v;
		cin >> u >> v;
		graph[u].push_back(v);
		graph[v].push_back(u);
	}

	for (int i = 1; i <= N; i++)
		sort(graph[i].begin(), graph[i].end());
	DFS(s);
	cout << '\n';
	memset(vis, 0, sizeof(vis));
	BFS();

	return 0;
}

File: 백준/Silver/1260. DFS와 BFS/DFS와 BFS.java
import java.io.*;
import java.util.*;

public class Main {
    static int N, M, V;
    static ArrayList<Integer>[] adj;
    static StringBuilder sb;
    static boolean[] vis;

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());
        sb = new StringBuilder();

        N = Integer.parseInt(st.nextToken());
        M = Integer.parseInt(st.nextToken());
        V = Integer.parseInt(st.nextToken());

        adj = new ArrayList[N + 1];
        for (int i = 0; i <= N; i++) {
            adj[i] = new ArrayList<>();
        }

        for (int i = 0; i < M; i++) {
            st = new StringTokenizer(br.readLine());
            int a = Integer.parseInt(st.nextToken());
            int b = Integer.parseInt(st.nextToken());
            adj[a].add(b);
            adj[b].add(a);
        }

        for (int i = 0; i <= N; i++) {
            Collections.sort(adj[i]);
        }
        vis = new boolean[N + 1];
        DFS(V);
        sb.append("\n");
        vis = new boolean[N + 1];
        BFS();

        System.out.println(sb.toString());
    }

    static void DFS(int cur) {
        vis[cur] = true;
        sb.append(cur).append(" ");

        for (int next : adj[cur]) {
            if (vis[next]) continue;
            DFS(next);
        }
    }

    static void BFS() {
        Queue<Integer> q = new ArrayDeque<>();
        q.offer(V);
        vis[V] = true;
        
        while (!q.isEmpty()) {
            int cur = q.poll();
            sb.append(cur).append(" ");

            for (int next : adj[cur]) {
                if (vis[next]) continue;

                q.offer(next);
                vis[next] = true;
            }
        }
    }
}
